### PR TITLE
Removed the FileSystemDataRepository.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Cleaned up version of #1138.

Ready for review.

### CODE REVIEW

File: ga4gh/cli.py [hunk 1]
```diff
@@ -180,9 +180,10 @@ def __init__(self, args):
         # depending on the prefix.
         filePrefix = "file://"
         if args.baseUrl.startswith(filePrefix):
-            dataDir = args.baseUrl[len(filePrefix):]
-            theBackend = backend.Backend(
-                datarepo.FileSystemDataRepository(dataDir))
+            repoPath = args.baseUrl[len(filePrefix):]
+            repo = datarepo.SqlDataRepository(repoPath)
+            repo.open("r")
```
Comment (dcolligan) [inline ga4gh/cli.py]: MODE_READ

File: tests/end_to_end/test_client_json.py [hunk 2]
```diff
@@ -28,9 +28,10 @@ class TestClientOutput(unittest.TestCase):
     """
     def setUp(self):
         self._maxDiff = None
-        self._dataDir = paths.testDataDir
-        self._dataUrl = "file://{}".format(self._dataDir)
-        dataRepository = datarepo.FileSystemDataRepository(self._dataDir)
+        repoPath = paths.testDataRepo
+        self._dataUrl = "file://{}".format(repoPath)
+        dataRepository = datarepo.SqlDataRepository(repoPath)
+        dataRepository.open("r")
```
Comment (dcolligan) [inline tests/end_to_end/test_client_json.py]: MODE_READ

File: tests/unit/test_backends.py [hunk 3]
```diff
@@ -147,16 +148,14 @@ def testGetReferenceSetByIndexBadIndex(self):
...
+    Tests proper initialization of the SQL repo based on known
     files in the tests/data directory.
     """
     def setUp(self):
-        self._dataDir = os.path.join("tests", "data")
-        self._backend = backend.Backend(
-            datarepo.FileSystemDataRepository(self._dataDir))
-        self._dataRepo = self._backend.getDataRepository()
+        self._dataRepo = datarepo.SqlDataRepository(paths.testDataRepo)
+        self._dataRepo.open("r")
```
Comment (dcolligan) [inline tests/unit/test_backends.py]: MODE_READ

File: tests/unit/test_converters.py [hunk 4]
```diff
@@ -22,8 +22,8 @@ class TestSamConverter(unittest.TestCase):
     Tests for the GA4GH reads API -> SAM conversion.
     """
     def setUp(self):
-        dataRepository = datarepo.FileSystemDataRepository(
-            paths.testDataDir)
+        dataRepository = datarepo.SqlDataRepository(paths.testDataRepo)
+        dataRepository.open("r")
```
Comment (dcolligan) [inline tests/unit/test_converters.py]: MODE_READ

File: tests/unit/test_variant_annotations.py [hunk 5]
```diff
@@ -24,7 +26,8 @@ def _createVariantAnnotationSet(self, vcfDir):
         VCF files.
         """
         self._variantSetName = "testVariantSet"
-        self._repo = datarepo.FileSystemDataRepository("tests/data")
+        self._repo = datarepo.SqlDataRepository(paths.testDataRepo)
+        self._repo.open("r")
```
Comment (dcolligan) [inline tests/unit/test_variant_annotations.py]: MODE_READ

**Comment by dcolligan** [comment]
+1 after fixes

**Comment by jeromekelleher** [comment]
Thanks @dcolligan, I think I've clobbered all the ``open("r|w")`` instances.

**Comment by david4096** [comment]
Yay! Interesting to note that there wasn't really test coverage of this file 👍 

Just read through to verify server restart is still required when the data are changed.

**Comment by jeromekelleher** [comment]
At the moment a server restart is still required because of the clunky in-memory model. Once we've moved to a direct SQL interface for running queries though... this starts to get really useful! Online updates to your repo without server restarts, with the internal consistencies guaranteed by foreign key constraints.

File: ga4gh/cli.py [hunk 6]
```diff
@@ -1574,7 +1575,7 @@ def _updateRepo(self, func, *args, **kwargs):
         repo.
```
Comment (david4096) [inline ga4gh/cli.py]: typo "transatctional"

**Comment by david4096** [comment]
I pulled it down and ran against the example data downloaded using your script. I had to add the `repositoryVersion` key to the DB when receiving this error:

```
(server-1)➜  server-1 git:(remotefilesystem) ✗ python server_dev.py 
Traceback (most recent call last):
  File "server_dev.py", line 7, in <module>
    ga4gh.cli.server_main()
  File "/Users/david/server-1/ga4gh/cli.py", line 138, in server_main
    parsedArgs.config_file, parsedArgs.config, parsedArgs.port)
  File "/Users/david/server-1/ga4gh/frontend.py", line 233, in configure
    dataRepository.open("r")
  File "/Users/david/server-1/ga4gh/datarepo.py", line 314, in open
    self.load()
  File "/Users/david/server-1/ga4gh/datarepo.py", line 957, in load
    self._readSystemTable(cursor)
  File "/Users/david/server-1/ga4gh/datarepo.py", line 398, in _readSystemTable
    self._repositoryVersion = config["repositoryVersion"]
KeyError: u'repositoryVersion'
```

**Comment by jeromekelleher** [comment]
I'm not sure how that error happened @david4096, you must be using a DB produced by an older version of the code. The DB here is fine:
```
sqlite> .open tests/data/repo.db 
sqlite> select * from system;
schemaVersion|0.1
creationTimeStamp|2016-04-27 13:27:33
sqlite> 
```

File: ga4gh/cli.py (hunk 6 above)
Comment (jeromekelleher) [inline ga4gh/cli.py]: Done.

**Comment by jeromekelleher** [comment]
OK, we have 2 +1s here so I'm going to merge. I think the problem @david4096 must be a different issue, and these sorts of problems will go away when we don't have so many PRs in flight.

**Comment by diekhans** [comment]
Versioning the repo db might be useful in preventing such
errors.  We may find problems after release that requires an
incompatible schema change.  So a table with has metadata about
the repo db could contain a repo db version number.  The repo db
version number would only change if the db became incompatible,
not every code change.

**Comment by jeromekelleher** [comment]
This is already done @diekhans --- we have a `System` table which is a key-value store. One of the keys is ``schemaVersion``. This is a major.minor version number. If ``major != currentMajor``, raise an error. The idea is that we increment ``minor`` each time we have a (non-breaking) schema change, and ``major`` each time we have a breaking schema change.

I think David had a version of the code that was done before this, and hence the crash.

**Comment by david4096** [comment]
Agreed @jeromekelleher , just making sure.

**Comment by jeromekelleher** [comment]
I rebased this branch quite a bit as it was being reviewed and other branches were moving around underneath it. This is the cost of having lots of interdependent PRs open at once I guess. Hopefully everything is nice and linear now though.

**Pull request merged**